tools/clang/test/HLSLFileCheck/hlsl/types/vector/vec_size_annot.hlsl

```diff
@@ -0,0 +1,20 @@
+// RUN: %dxc -T ps_6_8 -E main -Qkeep_reflect_in_dxil -select-validator internal %s | FileCheck %s
+
+// Make sure the vector is annotated with vector size (DXIL 1.8 and higher)
+
+// CHECK: !dx.typeAnnotations
+
+// CHECK: !{i32 0, %"class.StructuredBuffer<MyStruct>" undef, !8, %struct.MyStruct undef, [[ANNOT1:![0-9]+]]}
+// CHECK: [[ANNOT1:![0-9]+]] = !{i32 16, [[ANNOT2:![0-9]+]]}
+// CHECK: [[ANNOT2:![0-9]+]] = !{i32 6, !"vec", i32 3, i32 0, i32 7, i32 9, i32 13, i32 4}
+
+struct MyStruct {
+    float4 vec;
+};
+
+StructuredBuffer<MyStruct> g_myStruct;
```

Comment:
Does this get applied to matrices as well? Maybe we could check that and maybe ensure it doesn't get applied to scalars?

Reply:
BTW: Matrix has a special annotation type already.

Reply:
I have added test case for matrix and scalar as well, and also SM 6.7 variant (the vector size annotation should not be present when compiled with SM 6.7.)